proof-systems/src/darlin/t_dlog_acc_marlin/test.rs

```diff
@@ -0,0 +1,410 @@
+            let a = G1::ScalarField::rand(rng);
+            let b = G1::ScalarField::rand(rng);
+            let mut c = a;
+            c.mul_assign(&b);
+            let mut d = c;
+            d.mul_assign(&b);
+
+            let circ_g1 = Circuit {
+                a: Some(a),
+                b: Some(b),
+                c: Some(c),
+                d: Some(d),
+                num_constraints,
+                num_variables,
+            };
+
+            let a_g2 = G2::ScalarField::rand(rng);
+            let b_g2 = G2::ScalarField::rand(rng);
+            let mut c_g2 = a_g2;
+            c_g2.mul_assign(&b_g2);
+            let mut d_g2 = c_g2;
+            d_g2.mul_assign(&b_g2);
+
+            let circ_g2 = Circuit {
+                a: Some(a_g2),
+                b: Some(b_g2),
+                c: Some(c_g2),
+                d: Some(d_g2),
+                num_constraints,
+                num_variables,
+            };
+
+            let (index_pk_g1, index_vk_g1) =
+                TDLogAccMarlin::<G1, G2, D>::circuit_specific_setup(&pc_pk_g1, circ_g1).unwrap();
+            let (index_pk_g2, index_vk_g2) =
+                TDLogAccMarlin::<G2, G1, D>::circuit_specific_setup(&pc_pk_g2, circ_g2).unwrap();
+
+            assert!(index_pk_g1.is_valid());
+            assert!(index_vk_g1.is_valid());
+            assert!(index_pk_g2.is_valid());
+            assert!(index_vk_g2.is_valid());
+
+            println!("Called index");
+
+            test_canonical_serialize_deserialize(true, &index_pk_g1);
+            test_canonical_serialize_deserialize(true, &index_vk_g1);
+            test_canonical_serialize_deserialize(true, &index_pk_g2);
+            test_canonical_serialize_deserialize(true, &index_vk_g2);
+
+            let dlog_acc = DualDLogItem::generate_random::<_, D>(rng, &pc_pk_g2, &pc_pk_g1);
+            let inner_sumcheck_acc = DualSumcheckItem::<G2, G1>::generate_random::<D>(
+                rng,
+                &index_pk_g2.index_vk.index,
+                &index_pk_g1.index_vk.index,
+                &pc_pk_g2,
+                &pc_pk_g1,
+            );
+            let (_, bullet_poly_g1) = dlog_acc.compute_poly();
+            let (_, t_poly_g1) = inner_sumcheck_acc
+                .compute_poly(&index_pk_g2.index_vk.index, &index_pk_g1.index_vk.index);
+
+            let proof = TDLogAccMarlin::<G1, G2, D>::prove(
+                &index_pk_g1,
+                &pc_pk_g1,
+                circ_g1,
+                &inner_sumcheck_acc,
+                &dlog_acc,
+                &t_poly_g1,
+                &bullet_poly_g1[0],
+                zk,
+                if zk { Some(rng) } else { None },
+            );
+
+            assert!(proof.is_ok());
+
+            let proof = proof.unwrap();
+
+            assert!(proof.is_valid());
+
+            println!("Called prover");
+
+            test_canonical_serialize_deserialize(true, &proof);
+
+            // Success verification
```

Comment:
Can we be more specific, when doing negative tests, on which part we expect to fail (doable when splitting the succinct check function into IOP verification and opening proof succinct verification) ? E.g. SuccinctIOP/Outer sumcheck, SuccinctIOP/inner sumcheck aggregation, HardIOP/check t poly commitment, SuccinctDLOG/failed succinct check, HardDLOG/failed hard check. I reckon that for some of these it might be difficult to craft data to make the procedure fail in some specific spots, but at least let's try to detail failing assertions for the already existing ones